src/pyodide/internal/asgi.py

```diff
@@ -99,53 +99,118 @@ async def send(got):
-            # body is corrupted. This is fine because the session is going to
-            # exit at this point.
-            px = create_proxy(got["body"])
+            # Check if this is a server-sent events response
+            for k, v in headers:
+                if k.lower() == "content-type" and v.lower().startswith(
+                    "text/event-stream"
+                ):
+                    is_sse = True
+
+                    # For SSE, create and return the response immediately after http.response.start
+                    resp = Response.new(
+                        readable, headers=Object.fromEntries(headers), status=status
+                    )
+                    result.set_result(resp)
+                    break
+
+        elif got["type"] == "http.response.body":
+            body = got["body"]
+            more_body = got.get("more_body", False)
+
+            # Convert body to JS buffer
+            px = create_proxy(body)
             buf = px.getBuffer()
             px.destroy()
-            resp = Response.new(
-                buf.data, headers=Object.fromEntries(headers), status=status
-            )
-            result.set_result(resp)
 
-    await app(request_to_scope(req, env), receive, send)
-    return await result
+            if is_sse:
+                # For SSE, write chunk to the stream
+                await writer.write(buf.data)
+                # If this is the last chunk, close the writer
+                if not more_body:
+                    await writer.close()
+                    finished_response.set()
+            else:
+                resp = Response.new(
+                    buf.data, headers=Object.fromEntries(headers), status=status
+                )
+                result.set_result(resp)
+                finished_response.set()
+
+    # Run the application in the background to handle SSE
+    async def run_app():
+        try:
+            await app(request_to_scope(req, env), receive, send)
+
+            # If we get here and no response has been set yet, the app didn't generate a response
+            if not result.done():
+                await writer.close()  # Close the writer
+                finished_response.set()
+                result.set_exception(
+                    RuntimeError("The application did not generate a response")
+                )
+        except Exception as e:
+            # Handle any errors in the application
+            if not result.done():
+                await writer.close()  # Close the writer
+                finished_response.set()
+                result.set_exception(e)
+
+    # Create task to run the application in the background
+    app_task = create_task(run_app())
+
+    # Wait for the result (the response)
+    response = await result
+
+    # For non-SSE responses, we need to wait for the application to complete
+    if not is_sse:
+        await app_task
```

Comment:
Should we `ctx.waitUntil(app_task)` in the `sse` case?

Reply:
I don't think so? Our SSE examples don't do that.